enclave/src/rpc/rpc_withdraw.rs

```diff
@@ -99,26 +101,53 @@ pub mod tests {
     use sp_core::Pair;
 
     pub fn test_given_valid_call_then_succeed() {
+        let rpc_gateway = Box::new(RpcGatewayMock::mock_withdraw(true));
+
         let top_extractor = Box::new(TrustedOperationExtractorMock {
             trusted_operation: Some(create_withdraw_order_operation()),
         });
 
-        let rpc_gateway = Box::new(RpcGatewayMock::mock_withdraw(true));
-
         let rpc_withdraw = RpcWithdraw::new(top_extractor, rpc_gateway);
 
         let result = rpc_withdraw.method_impl(create_dummy_request());
 
         assert!(result.is_ok());
     }
 
-    pub fn test_given_unauthorized_access_then_return_error() {
+    pub fn test_incrementing_nonce_wont_match() {
```

Comment:
This test is not clear to me. Could you either adapt the name (why would an incremented nonce not match?) or add some documentation?

Reply:
This test is of a nonce incrementing but trying to validate the same value as the previous, so it won`t match.

Reply:
I see. So it's more like "test_rejecting_outdated_nonce" or along this way?

Reply:
If so, could you also add a test that two requests of the same account with correct nonces will get accepted?

Reply:
Actually... That's not the right place for these tests is it? Here you're testing the mock not your actual function, am I wrong?

Reply:
Hm, yes you are right, this is not the best place for that.

Reply:
I think the tests should be placed somwhere here: https://github.com/Polkadex-Substrate/polkadexTEE-worker/blob/develop/enclave/src/rpc/polkadex_rpc_gateway.rs#L75

Reply:
Test moved to polkadex_rpc_gateway, but still using rpc_withdraw